internal/controller/power_monitor_internal.go

```diff
@@ -153,6 +153,23 @@ func openshiftPowerMonitorClusterResources(pmi *v1alpha1.PowerMonitorInternal, c
 	}
 }
 
+func openshiftPowerMonitorNamespacedResources(pmi *v1alpha1.PowerMonitorInternal, cluster k8s.Cluster) []client.Object {
+	oshift := pmi.Spec.OpenShift
+
+	if cluster != k8s.OpenShift || !oshift.Enabled {
+		return nil
+	}
+
+	res := []client.Object{}
```

Comment:
question: is it possible to return real objects instead of client.Object?

Reply:
The reason it returns client.Object is because openshiftnamespacedresources can in theory return a variety of different objects. If that is too generic, we can instead divide it into multiple different functions. That said, it does not change much because resourceReconcilers accepts a list of client.Object as well. A lot of the reusable functions used in the operator are generic and so accept type client.Object.